cpp/src/io/parquet/writer_impl.cu

```diff
@@ -2339,16 +2363,21 @@ writer::impl::impl(std::vector<std::unique_ptr<data_sink>> sinks,
     _dict_policy(options.get_dictionary_policy()),
     _max_dictionary_size(options.get_max_dictionary_size()),
     _max_page_fragment_size(options.get_max_page_fragment_size()),
-    _int96_timestamps(options.is_enabled_int96_timestamps()),
+    _int96_timestamps(options.is_enabled_int96_timestamps() and
+                      not options.is_enabled_write_arrow_schema()),
     _utc_timestamps(options.is_enabled_utc_timestamps()),
     _write_v2_headers(options.is_enabled_write_v2_headers()),
+    _write_arrow_schema(options.is_enabled_write_arrow_schema()),
     _sorting_columns(options.get_sorting_columns()),
     _column_index_truncate_length(options.get_column_index_truncate_length()),
     _kv_meta(options.get_key_value_metadata()),
     _single_write_mode(mode),
     _out_sink(std::move(sinks)),
     _compression_statistics{options.get_compression_statistics()}
 {
+  if (options.is_enabled_int96_timestamps() and options.is_enabled_write_arrow_schema()) {
```

Comment:
Warn if arrow schema and int96 timestamps are simultaneously enabled for now. int96 timestamp removal work will be in a separate near-future PR.